### PR TITLE
TokenService improvements

### PR DESCRIPTION
**Pull request by kumavis**
1. should throw if api returns a json error. api returns a 200 ok but with json error obj
we should check for that an elevate to an error
2. token lookup uri is wrong. fixed. (nice catch @NiranjanaBinoy !)
3. return types were wrong, fixed.
4. added fallback value for `safelyExecute` usage
5. removed unsound `safelyExecute` usage

i guess this is a breaking change bc types?

### CODE REVIEW

**Review by NiranjanaBinoy — APPROVED** [review]
LGTM

**Comment by kumavis** [comment]
we need this for every api endpoint, eh?

**Comment by kumavis** [comment]
is there a prettier bot that can fix this for me

**Comment by kumavis** [comment]
please verify my changes are sound around `safelyExecute`

File: src/apis/token-service.ts [hunk 1]
```diff
@@ -38,35 +44,36 @@ export async function fetchTokenList(chainId: string): Promise<Response> {
...
+
+/**
+ * Parse response
+ *
+ * @return Promise resolving request response json value
+ */
+async function parseJsonResponse(apiResponse: Response): Promise<unknown> {
+  const responseObj = await apiResponse.json();
+  // api may return errors as json without setting an error http status code
+  if (responseObj?.error) {
```
Comment (ryanml) [inline src/apis/token-service.ts:75]: nit: Is the `?` necessary given we are checking for the props existence in the first place?

Comment (kumavis) [reply src/apis/token-service.ts:75]: responseObj could be null, thats valid json
unlikely tho

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**